tests/Integration/inc/Engine/Common/PerformanceHints/Frontend/Subscriber/maybe_apply_optimizations.php

```diff
@@ -66,6 +72,10 @@ public function testShouldReturnAsExpected( $config, $expected ) {
 		}
 
 		Functions\when( 'wp_create_nonce' )->justReturn( '96ac96b69e' );
+		Functions\when( 'is_user_logged_in' )->justReturn( $config['is_logged_in'] ?? false );
```

Comment:
We should not be using mocking functions in integration tests, the WP integration suite offers helper functions to manage users